client/lib/purchases/assembler.js

```diff
@@ -17,9 +15,7 @@ function createPurchaseObject( purchase ) {
 		amount: Number( purchase.amount ),
 		attachedToPurchaseId: Number( purchase.attached_to_purchase_id ),
 		mostRecentRenewDate: purchase.most_recent_renew_date,
-		mostRecentRenewMoment: purchase.most_recent_renew_date
-			? i18n.moment( purchase.most_recent_renew_date )
-			: null,
+		mostRecentRenewDateValid: Boolean( purchase.most_recent_renew_date ),
```

Comment:
I'm not very happy about including derived values like this in the Redux state. Why not store just `mostRecentRenewDate` and let the consumers check the validity themselves?

If some computation is needed at multiple places, it can be a selector or a helper function.

Reply:
`renewDate` has some specific semantics, and to centralise the validity checks, I I added `renewDateValid`. It seemed like more of a data layer concern rather than something that should be pushed on to consumers, which would need to all be aware of the semantics.

For consistency, I added it to the other ones as well with the simpler checks that match existing code.

Reply:
> `renewDate` has some specific semantics

Some of these semantics seem to be just bugs that should be fixed in the server-side implementation of the REST endpoints. For example, if the renew date on a purchase is not set, its date string is `"0000-00-00 00:00:00"`. The PHP code parses and formats it with the `DateTime` PHP class, and the result sent in the REST response is `"-001-11-30T00:00:00+00:00"`. Note that month being set to zero means "December of the previous year", and similarly day set to zero means "last day of the previous month".